pkg/operator/ceph/cluster/cluster.go

```diff
@@ -397,6 +411,49 @@ func (c *ClusterController) preClusterStartValidation(cluster *cluster, clusterO
 	return nil
 }
 
+// Remove the default root=default and replicated_rule CRUSH objects which are created by Ceph on initial startup.
+// Those objects may interfere with the normal operation of the cluster.
+// Note that errors which indicate that the objects are in use are ignored and the objects will continue to exist in that case.
+func (c *cluster) removeDefaultCrushMap() error {
+	logger.Info("removing default objects, if applicable")
+	args := []string{"osd", "crush", "rule", "rm", "replicated_rule"}
+	cephCmd := client.NewCephCommand(c.context, c.Namespace, args)
```

Comment:
Use the context from `c` and the executor to run a command instead. I know it might be confusing because the code base is not homogenous yet...

Reply:
@leseb Thanks. I tried that, and in contrast to the client.NewCephCommand way, c.context.Executor will not auto-add the arguments required for the command to succeed, such as keyring and config paths.

I can of course do all that manually, but duplicating the work of https://github.com/rook/rook/blob/master/pkg/daemon/ceph/client/command.go#L56-L88 seems not ideal. Calling that function would of course also be an option.

Which way is the preferred one?

1. c.context.Executor + FinalizeCephCommandArgs + whatever else may be needed
2. c.context.Executor + pass only the strictly necessary args by composing them manually
3. stay with client.NewCephCommand

Reply:
Actually my bad, `NewCephCommand` should be used here since it's building the correct arguments. So please dismiss my earlier comment. The executor should be used for non-ceph commands only :).